### PR TITLE
[Merged by Bors] - feat: Port Data.PFunctor.Multivariate.W

### PR DESCRIPTION
**Pull request by casavaca**
---

- [x] depends on: #1835 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#1835
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Data/PFunctor/Multivariate/W.lean [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+theorem wpRec_eq {α : TypeVec n} {C : Type _}
+    (g : ∀ (a : P.A) (f : P.last.B a → P.last.W), P.WPath ⟨a, f⟩ ⟹ α → (P.last.B a → C) → C)
+    (a : P.A) (f : P.last.B a → P.last.W) (f' : P.WPath ⟨a, f⟩ ⟹ α) :
+    P.wpRec g ⟨a, f⟩ f' = g a f f' fun i => P.wpRec g (f i) (P.wPathDestRight f' i) := rfl
+set_option linter.uppercaseLean3 false in
+#align mvpfunctor.Wp_rec_eq MvPFunctor.wpRec_eq
+
+-- Note: we could replace Prop by Type _ and obtain a dependent recursor
+theorem Wp_ind {α : TypeVec n} {C : ∀ x : P.last.W, P.WPath x ⟹ α → Prop}
```
Comment (alexkeizer) [inline Mathlib/Data/PFunctor/Multivariate/W.lean:173]: Should this be `wp_ind`, to be consistent with the capitalization of `wp`?

Comment (casavaca) [reply Mathlib/Data/PFunctor/Multivariate/W.lean:173]: Yes. I'll fix it when the dependency is merged.

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#1835~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**